geometry-schema.json

```diff
@@ -1,91 +1,224 @@
+          "type": "array",
+          "minItems": 2,
+          "items": {
+            "type": "number"
+          }
         },
-        {
-            "title": "LineString",
-            "properties": {
-                "type": { "enum": [ "LineString" ] },
-                "coordinates": { "$ref": "#/definitions/lineString" }
-            }
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#LineString",
+      "title": "GeoJSON LineString",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "LineString"
+          ]
         },
-        {
-            "title": "MultiLineString",
-            "properties": {
-                "type": { "enum": [ "MultiLineString" ] },
-                "coordinates": {
-                    "type": "array",
-                    "items": { "$ref": "#/definitions/lineString" }
-                }
+        "coordinates": {
+          "type": "array",
+          "minItems": 2,
+          "items": {
+            "type": "array",
+            "minItems": 2,
+            "items": {
+              "type": "number"
             }
+          }
         },
-        {
-            "title": "Polygon",
-            "properties": {
-                "type": { "enum": [ "Polygon" ] },
-                "coordinates": { "$ref": "#/definitions/polygon" }
-            }
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#Polygon",
+      "title": "GeoJSON Polygon",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "Polygon"
+          ]
         },
-        {
-            "title": "MultiPolygon",
-            "properties": {
-                "type": { "enum": [ "MultiPolygon" ] },
-                "coordinates": {
-                    "type": "array",
-                    "items": { "$ref": "#/definitions/polygon" }
-                }
+        "coordinates": {
+          "type": "array",
+          "items": {
+            "type": "array",
+            "minItems": 4,
+            "items": {
+              "type": "array",
+              "minItems": 2,
+              "items": {
+                "type": "number"
+              }
             }
+          }
+        },
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
         }
-    ],
-    "definitions": {
-        "position": {
-            "description": "A single position",
+      }
+    },
+    {
+      "$id": "#MultiPoint",
```

Comment:
internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.